provisioning/service/src/Config/ParserUtility.cs

```diff
@@ -72,6 +72,23 @@ public static void EnsureBase64String(string input)
             /* Codes_SRS_PARSER_UTILITY_21_004: [The IsValidBase64 shall do nothing if the string is valid.] */
         }
 
+        /// <summary>
+        /// Helper to validate that the provided string is neither null nor white space.
+        /// </summary>
+        /// <param name="input">the <code>string</code> to be validated</param>
+        /// <exception cref="ArgumentException">if the provided <code>string</code> do not fits the criteria</exception>
+        public static void EnsureNotNullOrWhiteSpaceString(string input)
+        {
+            if (string.IsNullOrWhiteSpace(input))
+            {
+                /* Codes_SRS_PARSER_UTILITY_21_016: [The IsNullOrWhiteSpace shall throw ArgumentException if the provided string is null.] */
+                /* Codes_SRS_PARSER_UTILITY_21_017: [The IsNullOrWhiteSpace shall throw ArgumentException if the provided string is whitespace.] */
+                throw new ArgumentException("parameter cannot be null or whiteSpace");
```

Comment:
Please move the check within the function requiring it and throw `ArgumentNullException` passing the `nameof(argument)`. 
`parameter` is too generic and is not in-line with C# argument validation.